crates/primitives/src/transaction/mod.rs

```diff
@@ -305,50 +305,38 @@ impl Transaction {
         }
     }
 
-    // TODO: dedup with effective_tip_per_gas
-    /// Determine the effective gas limit for the given transaction and base fee.
+    /// Returns the effective miner gas tip cap (`gasTipCap`) for the given base fee:
+    /// `min(maxFeePerGas - baseFee, maxPriorityFeePerGas)`
+    ///
     /// If the base fee is `None`, the `max_priority_fee_per_gas`, or gas price for non-EIP1559
     /// transactions is returned.
     ///
-    /// If the `max_fee_per_gas` is less than the base fee, `None` returned.
-    pub fn effective_gas_tip(&self, base_fee: Option<u64>) -> Option<u128> {
-        if let Some(base_fee) = base_fee {
-            let max_fee_per_gas = self.max_fee_per_gas();
-
-            if max_fee_per_gas < base_fee as u128 {
-                None
-            } else {
-                let effective_max_fee = max_fee_per_gas - base_fee as u128;
-                Some(std::cmp::min(effective_max_fee, self.priority_fee_or_price()))
-            }
-        } else {
-            Some(self.priority_fee_or_price())
+    /// Returns `None` if the basefee is higher than the [Transaction::max_fee_per_gas].
+    pub fn effective_tip_per_gas(&self, base_fee: Option<u64>) -> Option<u128> {
+        if base_fee.is_none() {
+            // If base_fee is None, then return the priority fee or the gas price.
+            return Some(self.priority_fee_or_price());
         }
-    }
 
-    /// Returns the effective miner gas tip cap (`gasTipCap`) for the given base fee:
-    /// `min(maxFeePerGas - baseFee, maxPriorityFeePerGas)`
-    ///
-    /// Returns `None` if the basefee is higher than the [Transaction::max_fee_per_gas].
-    pub fn effective_tip_per_gas(&self, base_fee: u64) -> Option<u128> {
-        let base_fee = base_fee as u128;
+        // Convert the base fee to u128 if present, or provide a default value
+        let base_fee = base_fee.unwrap_or_default() as u128;
```

Comment:
can probably get rid of this `unwrap_or_default` if we match on the `base_fee` like this at the beginning of the method:
```rust
let base_fee = match base_fee {
    Some(base_fee) => base_fee
    None => return self.priority_fee_or_price()
}
```

Reply:
> can probably get rid of this `unwrap_or_default` if we match on the `base_fee` like this at the beginning of the method:
> 
> ```rust
> let base_fee = match base_fee {
>     Some(base_fee) => base_fee
>     None => return self.priority_fee_or_price()
> }
> ```

Thank you !

Changes has been made